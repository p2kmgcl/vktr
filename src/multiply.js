import isNumber from "./assertions/is-number";
import isVector from "./assertions/is-vector";
import map from "./map";

/**
 * Returns a new Vector result of (a * scalar)
 * @param {Vector} a
 * @param {number} scalar
 * @return {Vector}
 */
function multiply(a, scalar) {
  isVector(a);
  isNumber(scalar);

  return map(a, component => component * scalar);
}

export default multiply;
