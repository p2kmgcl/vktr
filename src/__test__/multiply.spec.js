import multiply from "../multiply";
import Vector from "../vector";

describe("multiply()", () => {
  it("multiplies every component by a number", () => {
    const a = new Vector(0, 1, 0, -2);

    expect(multiply(a, -4)).toMatchSnapshot();
  });

  it("requires a number parameter", () => {
    const a = new Vector(1);

    expect(() => multiply(a, null)).toThrow(TypeError);
    expect(() => multiply(a, "a")).toThrow(TypeError);
    expect(() => multiply(a, [1])).toThrow(TypeError);
    expect(() => multiply(a, NaN)).toThrow(TypeError);
  });
});
