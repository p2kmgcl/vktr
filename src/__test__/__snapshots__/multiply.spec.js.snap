// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiply() multiplies every component by a number 1`] = `
Array [
  -0,
  -4,
  -0,
  8,
]
`;
